### PR TITLE
fix: add agent exec abstraction

### PR DESCRIPTION
**Pull request by sreya**
This PR adds an `agentexec.Execer` to circumvent a bug I introduced that surfaces when trying to test `agent` subpackages. There were some obvious code smells with reading environment variables past startup but I thought `agentexec` was contained enough that I could get away with it...unfortunately enabling it in dogfood and then also trying to run the `agent/*` suite of tests results in a ton of problems.

I'm not really sure if there's a more ergonomic way to enable this API but I'm all ears if someone has a better alternative.

### CODE REVIEW

File: scripts/rules.go [hunk 1]
```diff
@@ -503,7 +503,7 @@ func noExecInAgent(m dsl.Matcher) {
 				!m.File().PkgPath.Matches("/agentexec") &&
 				!m.File().Name.Matches(`_test\.go$`),
 		).
-		Report("The agent and its subpackages should not use exec.Command or exec.CommandContext directly. Consider using agentexec.CommandContext instead.")
+		Report("The agent and its subpackages should not use exec.Command or exec.CommandContext directly. Consider using agentexec.Execer instead.")
```
Comment (deansheather) [inline scripts/rules.go:506]:
```suggestion
		Report("The agent and its subpackages should not use exec.Command or exec.CommandContext directly. Consider using an agentexec.Execer instead.")
```

File: scripts/rules.go [hunk 2]
```diff
@@ -521,5 +521,5 @@ func noPTYInAgent(m dsl.Matcher) {
 				!m.File().PkgPath.Matches(`/agentexec`) &&
 				!m.File().Name.Matches(`_test\.go$`),
 		).
-		Report("The agent and its subpackages should not use pty.Command or pty.CommandContext directly. Consider using agentexec.PTYCommandContext instead.")
+		Report("The agent and its subpackages should not use pty.Command or pty.CommandContext directly. Consider using a agentexec.Execer instead.")
```
Comment (deansheather) [inline scripts/rules.go:524]:
```suggestion
		Report("The agent and its subpackages should not use pty.Command or pty.CommandContext directly. Consider using an agentexec.Execer instead.")
```

**Review by deansheather — APPROVED** [review]
Looks good, I actually like this pattern more honestly

File: agent/agentexec/exec.go [hunk 3]
```diff
@@ -20,60 +20,101 @@ const (
...
+		return nil, xerrors.Errorf("get executable: %w", err)
 	}
 
 	bin, err := filepath.EvalSymlinks(executable)
 	if err != nil {
-		return "", nil, xerrors.Errorf("eval symlinks: %w", err)
+		return nil, xerrors.Errorf("eval symlinks: %w", err)
+	}
+
+	oomScore, ok := envValInt(EnvProcOOMScore)
```
Comment (mafredri) [inline agent/agentexec/exec.go:62]: I see there's no validation, but what happens if this is set to <-20 or >20, will everything fail to exec, will it silently be ignored, or something else?

Also not sure if system level prio max/min can be adjusted, typically it's 20 anyway. 🤔

**Review by mafredri — APPROVED** [review]
Clean implementation, nice job 👍🏻

File: agent/agentexec/exec.go (hunk 3 above)
Comment (sreya) [reply agent/agentexec/exec.go:62]: It'll fail to adjust the priority which will print a message to stderr but it won't fatally kill the command

**Pull request merged**